### PR TITLE
Create service directory in common layer

### PR DESCRIPTION
**Pull request by DelphineChiu**
Summary:
1. Create service directory in common layer
   - Move host/ipmb/ipmi/sensor/usb services to service folder
   - Move main.c to service folder
2. Fix warning messages

Test Plan:
1. Build Code: pass

### CODE REVIEW

File: meta-facebook/yv35-bb/CMakeLists.txt [hunk 1]
```diff
@@ -7,10 +7,10 @@ project(yv35-bb)
 
 set(common_path ../../common)
 FILE(GLOB app_sources src/ipmi/*.c src/platform/*.c src/lib/*.c )
-FILE(GLOB common_sources ${common_path}/*.c ${common_path}/ipmi/*.c ${common_path}/host/*.c ${common_path}/sensor/*.c ${common_path}/usb/*.c ${common_path}/hal/*.c ${common_path}/dev/*.c ${common_path}/lib/*.c ${common_path}/ipmb/*.c)
+FILE(GLOB common_sources ${common_path}/service/*.c ${common_path}/service/*/*.c ${common_path}/hal/*.c ${common_path}/dev/*.c ${common_path}/lib/*.c ${common_path}/shell/*.c)
```
Comment (GoldenBug) [inline meta-facebook/yv35-bb/CMakeLists.txt:10]: I think we still need these 2 to build?
```
${common_path}/pldm/*.c 
${common_path}/mctp/*.c
```

File: meta-facebook/yv35-cl/CMakeLists.txt [hunk 2]
```diff
@@ -7,10 +7,10 @@ project(yv35-cl)
 
 set(common_path ../../common)
 FILE(GLOB app_sources src/ipmi/*.c src/platform/*.c src/lib/*.c)
-FILE(GLOB common_sources ${common_path}/*.c ${common_path}/ipmi/*.c ${common_path}/host/*.c ${common_path}/sensor/*.c ${common_path}/usb/*.c ${common_path}/hal/*.c ${common_path}/dev/*.c ${common_path}/lib/*.c ${common_path}/ipmb/*.c ${common_path}/shell/*.c)
+FILE(GLOB common_sources ${common_path}/service/*.c ${common_path}/service/*/*.c ${common_path}/hal/*.c ${common_path}/dev/*.c ${common_path}/lib/*.c ${common_path}/shell/*.c)
```
Comment (GoldenBug) [inline meta-facebook/yv35-cl/CMakeLists.txt:10]: I think we still need these 2 to build?
```
${common_path}/pldm/*.c 
${common_path}/mctp/*.c
```

File: meta-facebook/yv35-bb/CMakeLists.txt [hunk 3]
```diff
@@ -7,10 +7,10 @@ project(yv35-bb)
...
 
 target_sources(app PRIVATE ${app_sources})
 target_sources(app PRIVATE ${common_sources})
 target_include_directories(app PRIVATE ${ZEPHYR_BASE}/include/portability)
-target_include_directories(app PRIVATE ${common_path} ${common_path}/dev/include ${common_path}/ipmi/include ${common_path}/host ${common_path}/sensor ${common_path}/hal ${common_path}/dev ${common_path}/lib ${common_path}/usb ${common_path}/ipmb)
+target_include_directories(app PRIVATE ${common_path} ${common_path}/service/ipmi/include ${common_path}/service/host ${common_path}/service/sensor ${common_path}/service/usb ${common_path}/service/ipmb ${common_path}/hal ${common_path}/dev/include ${common_path}/lib ${common_path}/shell)
```
Comment (GoldenBug) [inline meta-facebook/yv35-bb/CMakeLists.txt:15]: I think we need these 2 to build still?
```
${common_path}/pldm 
${common_path}/mctp
```

File: meta-facebook/yv35-cl/CMakeLists.txt [hunk 4]
```diff
@@ -7,10 +7,10 @@ project(yv35-cl)
...
 
 target_sources(app PRIVATE ${app_sources})
 target_sources(app PRIVATE ${common_sources})
 target_include_directories(app PRIVATE ${ZEPHYR_BASE}/include/portability)
-target_include_directories(app PRIVATE ${common_path} ${common_path}/dev/include ${common_path}/ipmi/include ${common_path}/host ${common_path}/sensor ${common_path}/hal ${common_path}/dev ${common_path}/lib ${common_path}/usb ${common_path}/ipmb ${common_path}/shell)
+target_include_directories(app PRIVATE ${common_path} ${common_path}/service/ipmi/include ${common_path}/service/host ${common_path}/service/sensor ${common_path}/service/usb ${common_path}/service/ipmb ${common_path}/hal ${common_path}/dev/include ${common_path}/lib ${common_path}/shell)
```
Comment (GoldenBug) [inline meta-facebook/yv35-cl/CMakeLists.txt:15]: I think we need these 2 to build still?
```
${common_path}/pldm 
${common_path}/mctp
```

**Review by GoldenBug — COMMENTED** [review]
We'll need to add these once we rebase

**Comment by GoldenBug** [comment]
Or move them to services?

File: meta-facebook/yv35-bb/CMakeLists.txt (hunk 1 above)
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-bb/CMakeLists.txt:10]: We have fixed this and do the forced update.
Please review it. Thanks.

File: meta-facebook/yv35-cl/CMakeLists.txt (hunk 2 above)
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-cl/CMakeLists.txt:10]: We have fixed this and do the forced update.
Please review it. Thanks.

File: meta-facebook/yv35-bb/CMakeLists.txt (hunk 3 above)
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-bb/CMakeLists.txt]: We have fixed this and do the forced update.
Please review it. Thanks.

File: meta-facebook/yv35-cl/CMakeLists.txt (hunk 4 above)
Comment (RenChen-wiwynn) [reply meta-facebook/yv35-cl/CMakeLists.txt]: We have fixed this and do the forced update.
Please review it. Thanks.

**Pull request closed without merging**